lua/wikis/commons/Widget/EmptyPagePreview/Team.lua

```diff
@@ -0,0 +1,454 @@
+		return tournament.region
+	end)
+	local regionGroups = Array.groupBy(regions, FnUtil.identity)
+	Array.sortInPlaceBy(regionGroups, function(regionGroup) return - Table.size(regionGroup) end)
+	return (regionGroups[1] or {})[1]
+end
+
+---@private
+---@return Widget[]
+function EmptyTeamPagePreview:_rosterFromTransfers()
+	return WidgetUtil.collect(
+		HtmlWidgets.H3{children = 'Roster'},
+		HtmlWidgets.H4{children = 'Active'},
+		SquadAuto.active{
+			team = self.team,
+			roles = 'None,Loan,Substitute,Trial,Stand-in,Uncontracted', -- copied from commons template
+			type = 'Player_active',
+		},
+		HtmlWidgets.H4{children = 'Inactive'},
+		SquadAuto.inactive{
+			team = self.team,
+			type = 'Player_inactive',
+		},
+		HtmlWidgets.H4{children = 'Former'},
+		SquadAuto.former{
+			team = self.team,
+			roles = 'None,Loan,Substitute,Inactive,Trial,Stand-in,Uncontracted', -- copied from commons template
+			type = 'Player_former',
+		},
+		HtmlWidgets.H3{children = 'Active Organization'},
+		SquadAuto.active{
+			team = self.team,
+			not_roles = 'None,Loan,Substitute,Inactive,Trial,Stand-in,Uncontracted', -- copied from commons template
+			type = 'Organization_active',
+			title = 'Organization',
+			position = 'Position',
+		}
+	)
+end
+
+---@private
+---@return (Widget|Html)[]
+function EmptyTeamPagePreview:_matches()
+	return {
+		HtmlWidgets.H3{children = 'Most Recent Matches'},
+		MatchTable.results{
+			tableMode = 'team',
+			showType = true,
+			team = self.team,
+			limit = 10,
+		}
+	}
+end
+
+---@private
+---@return (Widget|Html)[]
+function EmptyTeamPagePreview:_results()
+	return {
+		HtmlWidgets.H3{children = 'Notable Results'},
+		ResultsTable.results{
+			team = self.team,
+			showType = true,
+			gameIcons = true,
+			awards = false,
+			achievements = true,
+			playerResultsOfTeam = false,
+			querytype = 'team',
+		}
+	}
+end
+
+---@private
+---@return {coaches: {flag: string?, displayName: string, pageName: string}[], nationalities: table<string, integer>}
+function EmptyTeamPagePreview:_getNationalitiesAndCoachesFromLastPlacement()
+	local data = self:_getPlayersAndCoachesFromLastPlacement()
+	local nationalities = {}
+
+	Array.forEach(data.players, function(player)
+		local flag = player.flag
+		if not flag then return end
+		nationalities[flag] = (nationalities[flag] or 0) + 1
+	end)
+
+	return {coaches = data.coaches, nationalities = nationalities}
+end
+
+---@private
+---@return {coaches: {flag: string?, displayName: string, pageName: string, name: string?}[],
+---players: {flag: string?, displayName: string, pageName: string, name: string?}[], startDate: string?}
+function EmptyTeamPagePreview:_getPlayersAndCoachesFromLastPlacement()
+	local latestResult = self:_fetchPlacements{limit = 1}[1]
+	if not latestResult then return {coaches = {}, nationalities = {}} end
+
+	local parsePerson = function (prefix)
+		local person = Page.pageifyLink(latestResult.opponentplayers[prefix])
```

Comment:
Shouldn't the name in opponentplayers already be pageified?

Reply:
depends on the wiki
i think for most not yet